src/serializers/encode_context.c

```diff
@@ -44,59 +42,38 @@ inline void GraphEncodeContext_SetEncodeState(GraphEncodeContext *ctx, EncodeSta
 
 inline uint64_t GraphEncodeContext_GetKeyCount(const GraphEncodeContext *ctx) {
 	assert(ctx);
-	return ctx->meta_keys_count + 1;
+	return raxSize(ctx->meta_keys) + 1;
 }
 
-void GraphEncodeContext_SetMetaKeysCount(GraphEncodeContext *ctx, uint64_t meta_keys_count) {
+inline void GraphEncodeContext_AddMetaKey(GraphEncodeContext *ctx, const char *key) {
 	assert(ctx);
-	ctx->meta_keys_count = meta_keys_count;
+	raxInsert(ctx->meta_keys, (unsigned char *)key, strlen(key), NULL, NULL);
 }
 
-inline uint64_t GraphEncodeContext_GetProcessedKeyCount(const GraphEncodeContext *ctx) {
-	assert(ctx);
-	return ctx->keys_processed;
-}
-
-inline uint64_t GraphEncodeContext_GetProcessedNodesCount(const GraphEncodeContext *ctx) {
-	assert(ctx);
-	return ctx->processed_nodes;
-}
-
-inline void GraphEncodeContext_SetProcessedNodesCount(GraphEncodeContext *ctx, uint64_t nodes) {
+inline unsigned char **GraphEncodeContext_GetMetaKeys(const GraphEncodeContext *ctx) {
 	assert(ctx);
-	ctx->processed_nodes = nodes;
+	return raxKeys(ctx->meta_keys);
```

Comment:
Just want to make sure `raxKeys` returns a null terminated strings?

Reply:
yes